src/client/display.c

```diff
@@ -474,6 +475,36 @@ display_port(struct writer *w, lldpctl_atom_t *port, int details)
 }
 
 static void
+display_local_ttl(struct writer *w, lldpctl_conn_t *conn, int details)
+{
+	char *ttl;
+	long int tx_hold;
+	long int tx_interval;
+
+	lldpctl_atom_t *configuration;
+	configuration = lldpctl_get_configuration(conn);
+	if (!configuration) {
+		log_warnx("lldpctl", "not able to get configuration. %s",
+		    lldpctl_last_strerror(conn));
+		return;
+	}
+
+	tx_hold = lldpctl_atom_get_int(configuration, lldpctl_k_config_tx_hold);
+	tx_interval = lldpctl_atom_get_int(configuration, lldpctl_k_config_tx_interval);
+
+	if (asprintf(&ttl, "%lu", tx_hold*tx_interval) == -1) {
+		log_warnx("lldpctl", "not enough memory to build TTL.");
```

Comment:
It's unlikely we ever hit this path, but it should have a goto to `lldpctl_atom_dec_ref()` at the end of the function or we will crash when using `ttl`.

Reply:
Oh dear ... good catch.